### ISSUE TITLE
Uncaught TypeError: Cannot set property 'sequence' of undefined

### ISSUE THREAD

**Issue by jakehockey10** [jlmakes/scrollreveal.js#269]
I looked around for an issue similar to this one and I couldn't find anything.

- I'm using Ubuntu 16.04
- I'm using Chrome 51.0.2704.106 (Official Build) (64-bit)
- I'm using ScrollReveal 3.1.4
- I load the page and I get the following error message:

```
Uncaught TypeError: Cannot set property 'sequence' of undefined
```
![screenshot from 2016-07-05 20-53-27](https://cloud.githubusercontent.com/assets/1821383/16606136/8ff739e0-42f2-11e6-8f80-34276a589903.png)

- I'm using Ruby on Rails, so my code invoking ScrollReveal looks like this:
```js
$(document).on('turbolinks:load', function () {
  [...]

  // Initialize and Configure Scroll Reveal Animation
  window.sr = ScrollReveal();
  sr.reveal('.sr-icons', {
    duration: 600,
    scale: 0.3,
    distance: '0px'
  }, 200);
  sr.reveal('.sr-button', {
    duration: 1000,
    delay: 200
  });
  sr.reveal('.sr-contact', {
    duration: 600,
    scale: 0.3,
    distance: '0px'
  }, 300);

  [...]
});
```
When debugging this, everything seems to be going great until `sr.store` is used to look up something, in which case it doesn't find what it is looking for because `sr.store` is empty at that time (not null).  Here is where that lookup fails to find what it is looking for:
![screenshot from 2016-07-05 20-59-18](https://cloud.githubusercontent.com/assets/1821383/16606225/6465db00-42f3-11e6-8294-5872b90eb47a.png)

This is the order in which I load my javascript assets in application.js:
```
//= require jquery
//= require jquery.easing
//= require jquery_ujs
//= require bootstrap-sprockets
//= require scrollreveal
//= require jquery.magnific-popup
//= require static_pages
//= require turbolinks
```
and the code above invoking ScrollReveal lies in static_pages.js.  scrollreveal.js is in my vendor/assets/javascripts/ folder.

Can I get some help getting pointed in the right direction please?  From what I can tell, the only place where `sr.store` get updated is in the function `_updateStore` and that doesn't get called but one place and that's after this error shows itself.

**Comment by jlmakes** [comment]
#### Why the heck?

I took a cursory look at Turbo Links, and I think this is related to your problem.

You are instantiating a brand new ScrollReveal on each page load, e.g. `window.sr = ScrollReveal();`, which means `sr.store` gets nuked and any remaining `data-sr-id` attributes in your DOM are now stale.

#### Suggestions

I would instantiate ScrollReveal only once, outside of your `turbolinks:load` function, and everything else should work as expected.

#### Addendum

Keep in mind, you can use `sr.sync()` to re-call all previous `reveal()` calls, which in essence will capture any new elements added to the DOM that match the selectors you used in those reveal calls.

**Note:** A related bug (#268) was _just_ fixed, so please upgrade to `3.1.5` for this to work correctly.

_For example:_

```js
window.sr = ScrollReveal()

sr.reveal('.sr-icons', {
  duration: 600,
  scale: 0.3,
  distance: '0px'
}, 200)

sr.reveal('.sr-button', {
  duration: 1000,
  delay: 200
})

sr.reveal('.sr-contact', {
  duration: 600,
  scale: 0.3,
  distance: '0px'
}, 300)

$(document).on('turbolinks:load', function () {
  sr.sync();
})
```

**Comment by jakehockey10** [comment]
@jlmakes thanks for your response.  I was wondering that myself, but I was thrown off that track when I was only hitting the instantiation once when I put a breakpoint there in DevTools.  Regardless, I gave your suggestion a shot and I get the warnings of no elements being found:

![screenshot from 2016-07-06 19-58-58](https://cloud.githubusercontent.com/assets/1821383/16640292/187b7ab2-43b4-11e6-9540-44862bf3d0d2.png)

with my code now looking like this:

```
// Initialize and Configure Scroll Reveal Animation
window.sr = ScrollReveal();
sr.reveal('.sr-icons', {
  duration: 600,
  scale: 0.3,
  distance: '0px'
}, 200);
sr.reveal('.sr-button', {
  duration: 1000,
  delay: 200
});
sr.reveal('.sr-contact', {
  duration: 600,
  scale: 0.3,
  distance: '0px'
}, 300);

$(document).on('turbolinks:load', function () {
  
  [...]

  // ScrollReveal
  sr.sync();

});
```

I'll keep trying to provide more useful information, and thank you again for taking a look at this with me.

**Comment by jakehockey10** [comment]
Solved!  I was introduced to your library through a template and that template had preset the `data-sr-id="[...]"` html tag attributes on the items the were being revealed by scrolling to them.  After I read a little bit more carefully how the code is written, I realized that at least this version of the library creates those attributes for you.  This is what was making the library assume that the store was NOT empty when it really was.  After removing the attributes from my html, ScrollReveal worked flawlessly.  I kept the instantiation of `window.sr` outside the `turbolinks:load` event but kept the `sr.reveal(...)` calls remain inside.

Thank you for the time you spent on this with me.
